src/gui/Layout.cpp

```diff
@@ -352,117 +338,17 @@ XojPageView* Layout::getViewAt(int x, int y)
-// 	int rb = rit - this->sizeRow.begin();	//get index
-// 	auto cit = std::lower_bound( this->sizeCol.begin(),  this->sizeCol.end(), x);
-// 	int cb = cit - this->sizeCol.begin();
-
-	auto t0 = Clock::now();
-
-	auto fast_linear_search = [](std::vector<unsigned> const& container, size_t start, size_t value) {
-		auto row_i = std::next(begin(container), start);
-		if (row_i != begin(container) && value < *row_i)
-		{
-			//Todo: replace with c++14
-			//Todo: rend(this->heightRows)
-			//Todo: std::make_reverse_iterator(c_i)
-			auto rbeg_i = std::reverse_iterator<decltype(row_i)>(row_i);
-			return std::find_if(rbeg_i, container.rend(), [value](int item) { return value > item; }).base();
-		}
-		return std::find_if(row_i, end(container), [value](int item) { return value <= item; });
-	};
-
-	auto const& row_i = fast_linear_search(this->heightRows, this->lastGetViewAtRow, y);
-	auto const& col_i = fast_linear_search(this->widthCols, this->lastGetViewAtCol, x);
-
-
-	auto t1 = Clock::now();
-
-	/* Linear Up or Down Search from last position: */
-	/* This is over 7x faster than the above code  */
-
-	// Rows:
-	int r = std::max(0, static_cast<int>(this->lastGetViewAtRow) - 1);
-	if (r > 0 && y <= this->heightRows[r])  //search lower
-	{
-		for (r--; r >= 0; r--)
-		{
-			if (y > this->heightRows[r])
-			{
-				break;  // past region - it's back up one
-			}
-		}
-		r++;
-	}
-	else  //search higher
-	{
-		for (; r < this->heightRows.size(); r++)
-		{
-			if (y <= this->heightRows[r]) break;  // found region
-		}
-	}
-
 
-	//Now for columns:
-	int c = std::max(0, static_cast<int>(this->lastGetViewAtCol) - 1);
-	if (c > 0 && x <= this->widthCols[c])  //search lower
-	{
-		for (c--; c >= 0; c--)
-		{
-			if (x > this->widthCols[c])
-			{
-				break;  // past region
-			}
-		}
-		c++;
-	}
-	else
-	{
-		for (; c < this->widthCols.size(); c++)
-		{
-			if (x <= this->widthCols[c]) break;  // found region
-		}
-	}
-
-
-	auto t2 = Clock::now();
-
-	//Binary Search ... too much overhead makes this a slower option in our use case.
-	/* This is over 2x faster than the first code block but 3x slower than the second code block*/
+	//Binary Search:
 	auto rit = std::lower_bound(this->heightRows.begin(), this->heightRows.end(), y);
-	int rb = rit - this->heightRows.begin();  //get index
+	int foundRow = rit - this->heightRows.begin();  //get index
 	auto cit = std::lower_bound(this->widthCols.begin(), this->widthCols.end(), x);
-	int cb = cit - this->widthCols.begin();
+	int foundCol = cit - this->widthCols.begin();
 
-	auto t3 = Clock::now();
 
-	searchtime1 += std::chrono::duration_cast<std::chrono::nanoseconds>(t1 - t0).count();
-	searchtime2 += std::chrono::duration_cast<std::chrono::nanoseconds>(t2 - t1).count();
-	searchtime3 += std::chrono::duration_cast<std::chrono::nanoseconds>(t3 - t2).count();
-
-
-	if (plotindex == 0)
+	if (foundCol < this->widthCols.size() && foundRow < this->heightRows.size())
```

Comment:
redundant, remove this line, since it is checked in auto optionalPage = this->mapper.at({foundCol, foundRow});